app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java

```diff
@@ -1,53 +1,256 @@
+import android.hardware.camera2.CaptureRequest;
+import android.hardware.camera2.params.StreamConfigurationMap;
 import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.util.Base64;
+import android.util.Log;
+import android.util.Size;
+import android.view.Surface;
+import android.view.TextureView;
 import android.view.View;
 import android.widget.Button;
-import android.widget.ImageView;
 
-public class CreateMeetupPhotoActivity extends CreateMeetupActivity implements View.OnClickListener {
+import java.io.ByteArrayOutputStream;
+import java.util.Collections;
+
+
+public class CreateMeetupPhotoActivity extends CreateMeetupActivity {
+
 
-    static final int REQUEST_IMAGE_CAPTURE = 1;
+    CameraManager cameraManager;
+    int cameraFacing;
 
-    private ImageView mImageView;
-    private Button mNextButton;
+    TextureView.SurfaceTextureListener surfaceTextureListener;
+    Size previewSize;
+    String cameraId;
+    private HandlerThread backgroundThread;
+    private CameraDevice.StateCallback stateCallback;
+    private Handler backgroundHandler;
+    private CameraDevice cameraDevice;
+    private TextureView textureView;
+    private CaptureRequest captureRequest;
+    private CameraCaptureSession cameraCaptureSession;
+    private CaptureRequest.Builder captureRequestBuilder;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         initialize(R.layout.activity_create_meetup_photo);
 
-        mNextButton = findViewById(R.id.meetup_next_button);
-        mNextButton.setEnabled(false);
+        cameraManager = (CameraManager) getSystemService(Context.CAMERA_SERVICE);
+        cameraFacing = CameraCharacteristics.LENS_FACING_BACK;
 
-        mImageView = findViewById(R.id.image_view);
-        Button cameraButton = findViewById(R.id.take_photo);
-
-        cameraButton.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View v) {
-                dispatchTakePictureIntent();
+        Button captureButton = findViewById(R.id.capture_button);
+        captureButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Bitmap bmp = textureView.getBitmap();
+                String pic = getStringFromBitmap(bmp);
+                mMeetup.mPhoto = pic;
+                executeNext();
             }
         });
+
+        textureView = findViewById(R.id.camera_preview);
+        surfaceTextureListener = new TextureView.SurfaceTextureListener() {
+            @Override
+            public void onSurfaceTextureAvailable(SurfaceTexture surfaceTexture, int width, int height) {
+                setUpCamera();
+                openCamera();
+            }
+
+            @Override
+            public void onSurfaceTextureSizeChanged(SurfaceTexture surfaceTexture, int width, int height) {
```

Comment:
Is it necessary to write these functions?

Reply:
Apparently they are required

Reply:
Huh, alright then